### PR TITLE
MultiSocket support - code [1/2]

### PR DESCRIPTION
**Pull request by squall0gd**
Support for multi-socket platforms.

### CODE REVIEW

File: docs/operator.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Notes:
 described in [cluster setup instructions using `cmk cluster-init`][cluster-init-op-manual].
 - The [cluster setup instructions using manually created Pods][indvidual-pods-op-manual] should only be used if and 
 only if running `cmk cluster-init` fails for some reason.
+- ___(multiple socket platforms)___ Dataplane pool shouldn't be spread within numa nodes, so `--num-dp-cores` value should be lower than number of cores on single socket.
```
Comment (ConnorDoyle) [inline docs/operator.md]: This is misleading. A single CPU list shouldn't spread across numa nodes, but there's no reason a pool shouldn't.

Comment (flyingcougar) [inline docs/operator.md]: Agree.... i think that pool can spread across numa nodes, the question is about i.e. NIC presence on specific socket. Another case is how to spread cpu across sockets. Spread all pools on all sockets or try to be "packed" and spread only some pools ?

**Comment by squall0gd** [comment]
@ConnorDoyle well there is few approaches to meat our goal. This PR is just a proposition for handling multi-socket platforms and my idea was do it as simple as it could be. The main change here is change in discovering topology, which is required imho anyway(b/c our current approach is making code really unclear imo). 

You've got right about spreading datapool on few sockets. In that case I propose following approach:
- spread pools on few numa nodes
- store information about sockets in our configuration topology
- add option to isolate for selecting particular 

I will try to implement it and commit into this PR.

File: docs/operator.md (hunk 1 above)
Comment (squall0gd) [inline docs/operator.md]: Could we discover correct way to implement this multi socket support first and then fix documentation :P? I guess we don't need this information at all if we spread pools on few sockets ;)

**Comment by lmdaly** [comment]
Tested - works well for me. Naming suggestion for vertical, horizontal as I think those could be confusing for users. Maybe continuous and distributed?

**Comment by flyingcougar** [comment]
@lmdaly `packed` and `spread` ?

**Comment by squall0gd** [comment]
I'm going to prepare separate PR for documentation

**Comment by flyingcougar** [comment]
Agree, once the documentation is reviewed - will merge both PR's.

**Comment by swatisehgal** [comment]
I like mode names spread and packed. Another one that popped in my head was: scatter and compact.

File: docs/cli.md [hunk 2]
```diff
@@ -58,12 +59,19 @@ Options:
...
+                        packed and spread [default: packed].
+  --dp-mode=<mode>      Data plane core allocation mode. Possible modes:
+                        packed and spread [default: packed].
   --publish             Whether to publish reports to the Kubernetes
                         API server.
   --pull-secret=<name>  Name of secret used for pulling Docker images from
                         restricted Docker registry.
   --saname=<name>       ServiceAccount name to pass
                         [default: cmk-serviceaccount].
+  --socket-id=<num>     ID of socket where allocated core should come from.
```
Comment (ConnorDoyle) [inline docs/cli.md]: Please pardon the late review, but curious whether this option is necessary. Isn't it equivalent to creating multiple pools? One advantage to the previous structure was that topology was confined to the initialization phase which allowed isolate to be very simple to implement and test.

Comment (squall0gd) [inline docs/cli.md]: It's not exactly equivalent to creating multiple pools, b/c you don't have to use it. If you don't mind use one socket or another then you don't care about this option and CMK behaviour doesn't change. If you care which socket will be used (b/c your DPDK workload needs access to specific NIC) then you can use this option.

Previous approach was simpler, but current is not so different. Most changes are in storing and gathering platform topology. Gathering those topology was required in any approach, storing was separate call from @lmdaly.

**Comment by flyingcougar** [comment]
LGTM

**Pull request merged**